crates/memory-testing/run_test.sh

```diff
@@ -0,0 +1,20 @@
+# Move to the root of the repository
+cd "$(dirname "$0")"
+cd ../../
+
+BASE_DIR="./crates/memory-testing"
+
+mkdir -p $BASE_DIR/output
+rm $BASE_DIR/output/*
+
+cargo build -p memory-testing
```

Comment:
I think we only need this in the no-docker condition?

Reply:
At the moment this script is running only the `capture-dumps` binary in Docker, as that's the only one that requires special permissions for the memory dump. The `analyze-dumps` is run on the host system.

I think we could move the analyze binary to run inside docker too to save one compilation, but assuming this is being run inside a developer's repo, they probably already have the binaries built anyway.

Reply:
Got it, never mind in that case.